airflow/models/dag.py

```diff
@@ -195,7 +195,7 @@ def __init__(
     ):
         self.user_defined_macros = user_defined_macros
         self.user_defined_filters = user_defined_filters
-        self.default_args = default_args or {}
+        self.default_args = dict(default_args or {})
```

Comment:
Maybe do a deepcopy then, so we are sure that the references aren't mutated as well.
```suggestion
        import copy
        self.default_args = copy.deepcopy(default_args)
```
WDYT?

Reply:
Yeah, I think we can do a deepcopy instead. Will make that change!